docs/decisions/0047-azure-open-ai-connectors.md

```diff
@@ -0,0 +1,188 @@
+    A[SemanticKernel.Connectors.OpenAI] --> B[OpenAI 2.0.0-beta.*]
+    A --> C[Azure.OpenAI 1.0.0-beta.17]
+    D[SemanticKernel.Connectors.AzureOpenAI] --> E[Azure.AI.OpenAI 2.0.0-beta.*]
+```
+
+The new `Options` pattern we be used as an improvement as well as a measure to avoid breaking changes with the legacy APIs.
+
+Following this change the the `SemanticKernel.Connectors.OpenAI` a new connector will be created `SemanticKernel.Connectors.AzureOpenAI` for Azure OpenAI services, using the Azure SDK `Azure.AI.OpenAI 2.0.0-beta.*` with all new APIs using the options approach.
+
+### Phases of the transition
+
+- **Phase 1**: Add new OpenAI SDK APIs to the current OpenAI connector and keep the Azure OpenAI APIs using the last Azure SDK.
+- **Phase 2**:
+  - Create a new connector for Azure OpenAI services using the new Azure SDK
+  - Deprecate all Azure OpenAI APIs in the `OpenAI` connector pointing to new `AzureOpenAI` connector
+  - Remove Azure SDK dependency from the OpenAI connector.
+  - Add `AzureOpenAI` connector to the `Microsoft.SemanticKernel` meta package.
+- **Phase 3**: Deprecate all legacy `OpenAI APIs` in the `OpenAI` connector pointing to new `Options` APIs.
+- **Phase 4**: Remove all legacy APIs from the OpenAI connector.
+
+### Impact
+
+Pros:
+
+- Minimal breaking changes for developers using the current OpenAI connector.
+- Clear separation of concerns between OpenAI and Azure OpenAI connectors.
+
+Cons:
+
+- Since `SemanticKernel.Connectors.AzureOpenAI` and `SemanticKernel.Connectors.OpenAI` share a same dependency of different versions, both packages cannot be used in the same project and a strategy will be needed when deploying both connectors:
+- Added dependency for both `Azure OpenAI 1.0-beta17` and `OpenAI 2.0-beta1`.
+
+### Dependency Management Strategies
+
+1. Use only one of the connectors in the same project, some modifications will be needed to accommodate `Concepts` and other projects that shares OpenAI and AzureOpenAI examples.
+2. Hold AzureOpenAI connector implementation until we are ready to break (exclude) all Azure APIs in OpenAI connector.
+3. Deploy a new project with a new namespace for `Azure.AI.OpenAI.Legacy 1.0.0-beta.17` and update our `SemanticKernel.Connectors.OpenAI` to use this new namespace to avoid version clashing on the `Azure.AI.OpenAI` namespace.
+
+## Option 2 - Independent Connectors from Start.
+
+This option is focused on creating fully independent connectors for OpenAI and Azure OpenAI services since the start with all breaking changes needed to achieve that.
+
+Impact:
+
+- `Azure` extension methods will be removed from `SemanticKernel.Connectors.OpenAI` namespace to avoid clashing with same names introduced in the new `SemanticKernel.Connectors.AzureOpenAI`.
+
+- All azure specific services like `AzureOpenAIChatCompletion`, `AzureOpenAITextGeneration` services will be obsoleted and throw not supported exceptions when used from `SemanticKernel.Connectors.OpenAI` connector.
```

Comment:
Is the intent here to provide a signal to developers that they need to add the new package dependency?